### PR TITLE
fix: publisher migration failures

### PR DESCRIPTION
**Pull request by theomonnom**
- fix publisher offer being sent before the SyncState (This caused the migration to fail/incorrect state)
- ~~add RtcConfiguration to the FFI server~~ another PR
- wait for subscriber pc on initial connection

### CODE REVIEW

File: livekit/src/rtc_engine/peer_transport.rs [hunk 1]
```diff
@@ -59,10 +59,7 @@ impl PeerTransport {
     }
 
     pub fn is_connected(&self) -> bool {
-        matches!(
-            self.peer_connection.ice_connection_state(),
-            IceConnectionState::Connected | IceConnectionState::Completed
-        )
+        self.peer_connection.connection_state() == PeerConnectionState::Connected
```
Comment (theomonnom) [inline livekit/src/rtc_engine/peer_transport.rs:62]: Now checking directly connection_state

File: livekit/src/room/mod.rs [hunk 2]
```diff
@@ -740,6 +792,8 @@ impl RoomSession {
 
         // unpublish & republish tracks
         let published_tracks = self.local_participant.tracks();
+
+        // Should I create a new task?
```
Comment (davidzhao) [inline livekit/src/room/mod.rs:756]: what's the intent behind creating a new task?

File: livekit/src/rtc_engine/mod.rs [hunk 3]
```diff
@@ -375,7 +451,7 @@ impl EngineInner {
     ) -> EngineResult<()> {
         let mut running_handle = self.running_handle.write().await;
         if running_handle.is_some() {
-            unreachable!("engine is already connected");
+            panic!("engine is already connected");
```
Comment (davidzhao) [inline livekit/src/rtc_engine/mod.rs:384]: is this the common way of error handling in Rust? would this crash the program?

**Review by davidzhao — APPROVED** [review]
lgtm!

File: livekit/src/rtc_engine/mod.rs (hunk 3 above)
Comment (theomonnom) [reply livekit/src/rtc_engine/mod.rs:384]: Yes, it's more like an assert here, this should never happens, gonna change the panic to assert

File: livekit/src/room/mod.rs (hunk 2 above)
Comment (theomonnom) [reply livekit/src/room/mod.rs:756]: The question is more like: do we want to wait for the track to be republished before finishing the reconnection logic? (Calling the reconnect event)

**Comment by theomonnom** [comment]
Now using channels instead of callbacks. Because in bad timing where the callbacks were not yet configured, we could have skipped certain events.

Channels will just buffer these events

File: livekit/src/room/mod.rs (hunk 2 above)
Comment (theomonnom) [reply livekit/src/room/mod.rs:756]: discussed in Slack

**Review by davidzhao — APPROVED** [review]

**Pull request merged**